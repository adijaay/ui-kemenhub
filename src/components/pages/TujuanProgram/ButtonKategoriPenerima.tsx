import { IconChevronRight } from "@tabler/icons-react";
import { useRouter } from "next/router";
import React from "react";

export default function ButtonKategoriPenerima() {
  const router = useRouter();

  const buttonHandler = () => {
    router.push("/kategori-penerima-program");
  };

  return (
    <button
      type="button"
      onClick={buttonHandler}
      className="relative flex items-center justify-between overflow-hidden rounded-xl bg-[#AF2A2D] px-4 py-3"
    >
      <img
        src="/decorations/vector-button-1.png"
        alt="vector-1"
        className="absolute right-0 top-13 w-[100px] rounded-br-xl"
      />
      <img
        src="/decorations/vector-button-2.png"
        alt="vector-2"
        className="absolute bottom-[27px] right-[35px] w-[100px] rounded-br-xl"
      />
      <img
        src="/decorations/vector-button-3.png"
        alt="vector-3"
        className="absolute left-[0px] top-[28px] w-[100px] rounded-bl-xl"
      />

      <div className="z-10 w-full text-start">
        <p className="text-xs font-normal leading-5 text-white">Baca Juga</p>
        <p className="text-sm font-bold leading-5 text-white">
          Kategori Penerima PIP
        </p>
      </div>
      <IconChevronRight size={24} color="#ffffff" stroke={2} />
    </button>
  );
}
