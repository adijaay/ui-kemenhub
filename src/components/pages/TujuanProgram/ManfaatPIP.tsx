import Container from "@/components/ui/Container";
import TextTitle from "@/components/ui/TextTitle";
import React from "react";

export default function ManfaatPIP() {
  return (
    <Container>
      <TextTitle>Manfaat Program</TextTitle>
      <ul className="flex list-disc flex-col gap-2 pl-4">
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Akses pendidikan merata sehingga semua peserta didik terdaftar dapat
          menyelesaikan pendidikan hingga tamat tingkat menengah.
        </li>
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Dukungan finansial untuk mengurangi beban biaya yang harus ditanggung
          oleh keluarga peserta didik.
        </li>
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Meningkatkan kesempatan bagi anak-anak untuk memperoleh pendidikan
          yang layak.
        </li>
      </ul>
    </Container>
  );
}
