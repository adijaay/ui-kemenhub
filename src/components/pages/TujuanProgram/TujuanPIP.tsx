import Container from "@/components/ui/Container";
import TextTitle from "@/components/ui/TextTitle";
import React from "react";

export default function TujuanPIP() {
  return (
    <Container>
      <TextTitle>Tujuan Program</TextTitle>
      <ul className="flex list-disc flex-col gap-2 pl-4">
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Mencegah siswa dari kemungkinan putus sekolah.
        </li>
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Mengajak siswa yang telah putus sekolah bisa kembali melanjutkan
          pendidikannya.
        </li>
        <li className="pl-2 text-sm font-normal text-[#1f1f1f]">
          Meringankan beban biaya pendidikan peserta didik, baik biaya langsung
          maupun tidak langsung.
        </li>
      </ul>
    </Container>
  );
}
