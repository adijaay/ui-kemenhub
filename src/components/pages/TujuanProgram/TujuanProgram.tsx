import Container from "@/components/ui/Container";
import Text from "@/components/ui/Text";
import React from "react";
import ManfaatPIP from "./ManfaatPIP";
import TujuanPIP from "./TujuanPIP";
import ButtonKategoriPenerima from "./ButtonKategoriPenerima";

export default function TujuanProgram() {
  return (
    <>
      <Container>
        <img src="/img/kartu-kip.png" alt="kartu-kip" />
        <Text className="text-justify">
          <span className="font-semibold">Program Indonesia Pintar (PIP)</span>{" "}
          ini dirancang untuk memastikan anak-anak usia sekolah dari keluarga
          miskin, rentan miskin, dan prioritas tetap mendapatkan akses
          pendidikan hingga tamat pendidikan menengah. Program ini mencakup
          jalur formal(SD hingga SMA/SMK dan jalur nonformal paket A hingga
          paket C, serta pendidikan khusus.
        </Text>
      </Container>

      <TujuanPIP />
      <ManfaatPIP />
      <ButtonKategoriPenerima />
    </>
  );
}
