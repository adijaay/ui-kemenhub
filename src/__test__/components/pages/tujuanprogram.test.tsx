import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { useRouter } from "next/router";
import TujuanProgram from "@/components/pages/TujuanProgram/TujuanProgram";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

const useRouterMock = useRouter as jest.Mock;

describe("Tujuan Program", () => {
  let push: jest.Mock;

  beforeEach(() => {
    push = jest.fn();
    useRouterMock.mockImplementation(() => ({
      push,
    }));
  });

  it("render tujuan program component correctly", () => {
    const page = render(<TujuanProgram />);
    expect(page).toMatchSnapshot();
  });
});
