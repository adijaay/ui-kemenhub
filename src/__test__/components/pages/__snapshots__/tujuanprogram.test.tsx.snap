// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tujuan Program render tujuan program component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <img
          alt="kartu-kip"
          src="/img/kartu-kip.png"
        />
        <p
          class="text-sm font-normal text-[#1f1f1f] text-justify"
        >
          Program Indonesia Pintar (PIP) dirancang untuk memastikan anak-anak usia sekolah dari keluarga miskin, rentan miskin, atau prioritas tetap mendapatkan akses pendidikan hingga tamat pendidikan menengah. Program ini mencakup jalur formal (SD hingga SMA/SMK) dan jalur non-formal (paket A hingga paket C), serta pendidikan khusus.
        </p>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <p
          class="font-semibold text-[#1f1f1f] undefined"
        >
          Tujuan Program Indonesia Pintar
        </p>
        <ul
          class="flex list-disc flex-col gap-2 pl-4"
        >
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Mencegah Putus Sekolah:
            </span>
             Program ini bertujuan untuk mencegah siswa dari kemungkinan putus sekolah.
          </li>
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Mengajak Kembali Siswa:
            </span>
             
            Diharapkan, siswa yang telah putus sekolah bisa kembali melanjutkan pendidikannya melalui PIP.
          </li>
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Meringankan Biaya Pendidikan:
            </span>
             
            PIP diharapkan dapat meringankan beban biaya pendidikan peserta didik, baik biaya langsung maupun tidak langsung.
          </li>
        </ul>
      </div>
      <div
        class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
      >
        <p
          class="font-semibold text-[#1f1f1f] undefined"
        >
          Manfaat Program Indonesia Pintar
        </p>
        <ul
          class="flex list-disc flex-col gap-2 pl-4"
        >
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Akses Pendidikan yang Merata:
            </span>
             
            Semua anak dari keluarga rentan dapat terus mengenyam pendidikan hingga tingkat menengah.
          </li>
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Dukungan Finansial:
            </span>
             Program ini memberikan bantuan untuk mengurangi beban biaya yang harus ditanggung oleh keluarga peserta didik.
          </li>
          <li
            class="pl-2 text-sm font-normal text-[#1f1f1f]"
          >
            <span
              class="font-semibold"
            >
              Peningkatan Kesempatan:
            </span>
             Dengan adanya PIP, kesempatan bagi anak-anak untuk memperoleh pendidikan yang layak semakin terbuka luas.
          </li>
        </ul>
      </div>
      <button
        class="relative flex items-center justify-between overflow-hidden rounded-xl bg-[#AF2A2D] px-4 py-3"
        type="button"
      >
        <img
          alt="vector-1"
          class="absolute right-0 top-13 w-[100px] rounded-br-xl"
          src="/decorations/vector-button-1.png"
        />
        <img
          alt="vector-2"
          class="absolute bottom-[27px] right-[35px] w-[100px] rounded-br-xl"
          src="/decorations/vector-button-2.png"
        />
        <img
          alt="vector-3"
          class="absolute left-[0px] top-[28px] w-[100px] rounded-bl-xl"
          src="/decorations/vector-button-3.png"
        />
        <div
          class="z-10 w-full text-start"
        >
          <p
            class="text-xs font-normal leading-5 text-white"
          >
            Baca Juga
          </p>
          <p
            class="text-sm font-bold leading-5 text-white"
          >
            Kategori Penerima PIP
          </p>
        </div>
        <svg
          class="tabler-icon tabler-icon-chevron-right "
          fill="none"
          height="24"
          stroke="#ffffff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 6l6 6l-6 6"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
    >
      <img
        alt="kartu-kip"
        src="/img/kartu-kip.png"
      />
      <p
        class="text-sm font-normal text-[#1f1f1f] text-justify"
      >
        Program Indonesia Pintar (PIP) dirancang untuk memastikan anak-anak usia sekolah dari keluarga miskin, rentan miskin, atau prioritas tetap mendapatkan akses pendidikan hingga tamat pendidikan menengah. Program ini mencakup jalur formal (SD hingga SMA/SMK) dan jalur non-formal (paket A hingga paket C), serta pendidikan khusus.
      </p>
    </div>
    <div
      class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
    >
      <p
        class="font-semibold text-[#1f1f1f] undefined"
      >
        Tujuan Program Indonesia Pintar
      </p>
      <ul
        class="flex list-disc flex-col gap-2 pl-4"
      >
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Mencegah Putus Sekolah:
          </span>
           Program ini bertujuan untuk mencegah siswa dari kemungkinan putus sekolah.
        </li>
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Mengajak Kembali Siswa:
          </span>
           
          Diharapkan, siswa yang telah putus sekolah bisa kembali melanjutkan pendidikannya melalui PIP.
        </li>
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Meringankan Biaya Pendidikan:
          </span>
           
          PIP diharapkan dapat meringankan beban biaya pendidikan peserta didik, baik biaya langsung maupun tidak langsung.
        </li>
      </ul>
    </div>
    <div
      class="flex flex-col gap-4 rounded-2xl border border-[#EAECF0] bg-white px-4 py-6 undefined"
    >
      <p
        class="font-semibold text-[#1f1f1f] undefined"
      >
        Manfaat Program Indonesia Pintar
      </p>
      <ul
        class="flex list-disc flex-col gap-2 pl-4"
      >
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Akses Pendidikan yang Merata:
          </span>
           
          Semua anak dari keluarga rentan dapat terus mengenyam pendidikan hingga tingkat menengah.
        </li>
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Dukungan Finansial:
          </span>
           Program ini memberikan bantuan untuk mengurangi beban biaya yang harus ditanggung oleh keluarga peserta didik.
        </li>
        <li
          class="pl-2 text-sm font-normal text-[#1f1f1f]"
        >
          <span
            class="font-semibold"
          >
            Peningkatan Kesempatan:
          </span>
           Dengan adanya PIP, kesempatan bagi anak-anak untuk memperoleh pendidikan yang layak semakin terbuka luas.
        </li>
      </ul>
    </div>
    <button
      class="relative flex items-center justify-between overflow-hidden rounded-xl bg-[#AF2A2D] px-4 py-3"
      type="button"
    >
      <img
        alt="vector-1"
        class="absolute right-0 top-13 w-[100px] rounded-br-xl"
        src="/decorations/vector-button-1.png"
      />
      <img
        alt="vector-2"
        class="absolute bottom-[27px] right-[35px] w-[100px] rounded-br-xl"
        src="/decorations/vector-button-2.png"
      />
      <img
        alt="vector-3"
        class="absolute left-[0px] top-[28px] w-[100px] rounded-bl-xl"
        src="/decorations/vector-button-3.png"
      />
      <div
        class="z-10 w-full text-start"
      >
        <p
          class="text-xs font-normal leading-5 text-white"
        >
          Baca Juga
        </p>
        <p
          class="text-sm font-bold leading-5 text-white"
        >
          Kategori Penerima PIP
        </p>
      </div>
      <svg
        class="tabler-icon tabler-icon-chevron-right "
        fill="none"
        height="24"
        stroke="#ffffff"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 6l6 6l-6 6"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
